### PR TITLE
Issue #20419: Servlet 6 disable ability to enable X-Powered-By

### LINKED ISSUE #20419: Servlet 6.0: Remove the recommendation that Servlet containers should include an `X-Powered-By`header.
Spec issue: `https://github.com/eclipse-ee4j/servlet-api/issues/272`

See "Appendix A: Change Log" here: https://github.com/eclipse-ee4j/servlet-api/blob/master/spec/src/main/asciidoc/servlet-spec-body.adoc

1) There is a test in the `com.ibm.ws.webcontainer.servlet.4.0_fat` bucket that tests this functionality `testServletXPoweredByHeader_Servlet50_Enabled`. Currently, the test is just disabled for Jakarta EE10 until we determine what to do with this work item. We also have a TODO in the Javadoc for the test that should be removed once we complete this issue.
2) At a minimum we need to update the WebContainer code to ensure the proper value for Servlet 6.0 is configured in the header if `disableXPoweredBy` is set to `false`.
3) Alternatively is there a way to prevent this configuration from being used in servlet-6.0+ since it isn't in the Servlet specification?
4) Once the WebContainer updates have been completed the test case in number 1 needs to be modified/enabled or a new test created to test whatever functionality is delivered.

The `com.ibm.ws.webcontainer.servlet.4.0` FAT bucket was repeated for Jakarta EE10 here: https://github.com/OpenLiberty/open-liberty/issues/20547

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #20419

### CODE REVIEW

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by posmith** [comment]
L2 message review not required

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -122,7 +122,7 @@ xPoweredBy.name=X-Powered-By
 xPoweredBy.desc=Alternative string for the X-Powered-By header setting. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.xpoweredby. There is no default value for this property. If the property is not set, the value of the X-Powered-By header is set to Servlet/<servlet spec version>, as defined by the Servlet specification.
 
 disableXPoweredBy.name=Disable the X-Powered-By header
-disableXPoweredBy.desc=Disable setting the X-Powered-By header. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.disablexpoweredby. For servlet-5.0 and newer Servlet features, the default value is true.
+disableXPoweredBy.desc=Disable setting the X-Powered-By header. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.disablexpoweredby. For servlet-5.0 and newer Servlet features, the default value is true. Only configurable for Servlet 5.0 or older.
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: I don't like how it says and newer servlet features since it's removed from servlet 6.0. 

Maybe reword to say: Only configurable for Servlet 5.0 or older. Default value is false for servlet-4.0 and older, but true for Servlet 5.0.

File: dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/publish/servers/servlet60_xPoweredByHeaderTest/server.xml [hunk 2]
```diff
@@ -8,12 +8,15 @@
...
-<server description="Server for testing Webcontainer">
+<server description="Server for test disableXPoweredBy configuration.r">
 
     <include location="../fatTestPorts.xml"/>
 
     <featureManager>
         <feature>servlet-6.0</feature>
     </featureManager>
+    
+     <webContainer disableXPoweredBy="false"/>
```
Comment (volosied) [inline dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/publish/servers/servlet60_xPoweredByHeaderTest/server.xml]: Should this be here? Or are you making sure that servlet 6.0 ignores this setting? Can you add a comment?

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: My intention here is that the behavior is still `true` for the `servlet-6.0` feature, it just can not be configured to false.

File: dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/publish/servers/servlet60_xPoweredByHeaderTest/server.xml (hunk 2 above)
Comment (pnicolucci) [reply dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/publish/servers/servlet60_xPoweredByHeaderTest/server.xml]: I've added a comment in the code review commit that I just pushed to this PR. This server.xml is used in the `test_XPoweredByCanNotBeEnabled` test.

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (volosied) [reply dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: I see. I think the current wording is fine then.

**Review by volosied — APPROVED** [review]
Looks good

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (pmd1nh) [reply dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: Only servlet-5.0 or older is configurable.

**Review by pmd1nh — APPROVED** [review]

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -122,7 +122,7 @@ xPoweredBy.name=X-Powered-By
 xPoweredBy.desc=Alternative string for the X-Powered-By header setting. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.xpoweredby. There is no default value for this property. If the property is not set, the value of the X-Powered-By header is set to Servlet/<servlet spec version>, as defined by the Servlet specification.
 
 disableXPoweredBy.name=Disable the X-Powered-By header
-disableXPoweredBy.desc=Disable setting the X-Powered-By header. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.disablexpoweredby. For servlet-5.0 and newer Servlet features, the default value is true.
+disableXPoweredBy.desc=Disable setting the X-Powered-By header. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.disablexpoweredby. For servlet-5.0 and newer Servlet features, the default value is true. Only configurable for Servlet 5.0 and older Servlet features.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: This is a little confusing, in part because the description doesn't tell you anything about the significance of setting the header, it just tells you what the property does, not why you might need to set it. The default per version is a little confusing too. I _think_ the following applies:
- in servlet-5.0 and later, by default no X-Powered-By header is set.
- in servlet 4.0 and earlier, the X-Powered-By header is set by default
- in servlet 5.0 and earlier, you can use this property to configure whether the X-Powered-By header is set. But in subsequent versions, you will not have the option to set an X-Powered-BY header.

if that's correct, then maybe something like:
```suggestion
disableXPoweredBy.desc=When this property is set to true,  the X-Powered-By header is not set. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the X-Powered-By header is set by default. This property is configurable only in servlet-5.0 and earlier. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.disablexpoweredby.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- made a few suggestions. Let me know if you have any questions. Thanks

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties (hunk 3 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties]: I've made the suggested changes and started a new personal build.

**Comment by pnicolucci** [comment]
#libby #build

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#libby
